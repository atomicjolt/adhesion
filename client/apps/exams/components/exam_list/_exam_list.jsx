import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import _ from 'lodash';
import moment from 'moment';
import canvasRequest from 'atomic-canvas/libs/action';
import { listQuizzesInCourse } from 'atomic-canvas/libs/constants/quizzes';
import appHistory from '../../history';
import Defines from '../../defines';
import * as ExamActions from '../../actions/exams';
import ExamListItem from './exam_list_item';
import { getTestingCentersAccount } from '../../actions/testing_centers_account';

const select = state => ({
  examList: state.exams.examList,
  lmsCourseId: state.settings.lms_course_id,
  toolConsumerInstanceName: state.settings.tool_consumer_instance_name,
  lmsUserId: state.settings.lms_user_id,
  examRequests: state.exams.examRequests,
});

export class BaseExamList extends React.Component {

  static propTypes = {
    canvasRequest: PropTypes.func.isRequired,
    getTestingCentersAccount: PropTypes.func.isRequired,
    loadExamRequests: PropTypes.func.isRequired,
    lmsCourseId: PropTypes.string.isRequired,
    examList: PropTypes.arrayOf(PropTypes.shape({})),
    toolConsumerInstanceName: PropTypes.string.isRequired,
    lmsUserId: PropTypes.string.isRequired,
    examRequests: PropTypes.shape({}),
  };

  static goToExam(id) {
    appHistory.push(`exams/${id}`);
  }

  static tableHeader(styles) {
    return (
      <tr>
        <th style={{ ...styles.th, ...styles.larger }}>
          EXAM
        </th>
        <th style={{ ...styles.th, ...styles.smaller }}>
          TESTING CENTER
        </th>
        <th style={{ ...styles.th, ...styles.smaller }}>
          STATUS
        </th>
      </tr>
    );
  }

  static getStyles() {
    return {
      header: {
        color: Defines.darkGrey,
        padding: '10px 20px 0px 20px',
      },
      message: {
        padding: '0px 20px',
        marginBottom: '20px',
      },
      hr: {
        borderTop: `2px solid ${Defines.lightGrey}`,
        margin: '0px',
      },
      table: {
        borderCollapse: 'collapse',
        width: '100%',
      },
      thead: {
        backgroundColor: Defines.lightBackground,
        color: Defines.lightText,
      },
      th: {
        fontWeight: 'normal',
        textAlign: 'left',
        padding: '15px 20px',
      },
      smaller: {
        width: '25%',
      },
      larger: {
        width: '50%',
      },
    };
  }

  componentWillMount() {
    const params = {
      course_id: this.props.lmsCourseId,
    };
    this.props.getTestingCentersAccount(this.props.toolConsumerInstanceName);
    this.props.canvasRequest(listQuizzesInCourse, params);
    this.props.loadExamRequests(this.props.lmsUserId);
  }

  filteredExams() {
    let exams = _.filter(this.props.examList, exam => (
      !this.props.examRequests[exam.id] ||
      (this.props.examRequests[exam.id] && this.props.examRequests[exam.id].status !== 'finished')
    ));
    exams = _.filter(exams, exam => _.includes(exam.access_code, 'proctored-exam'));
    exams = _.orderBy(exams, exam => moment(exam.due_at).toDate(), ['asc']);
    return exams;
  }

  examListItems() {
    return _.map(this.filteredExams(), exam => (
      <ExamListItem
        key={`exam_${exam.id}`}
        exam={exam}
        goToExam={id => BaseExamList.goToExam(id)}
        examRequest={this.props.examRequests[exam.id]}
      />
    ));
  }

  render() {
    if (!this.props.examRequests) { return null; }
    const styles = BaseExamList.getStyles();
    return (
      <div>
        <h1 style={styles.header}>Request an Exam</h1>
        <div style={styles.message}>
          If you need to change where you take an exam, create a new request.
        </div>
        <hr style={styles.hr} />
        <table className="qa-exam-list-table" style={styles.table}>
          <thead style={styles.thead}>
            {BaseExamList.tableHeader(styles)}
          </thead>
          <tbody>
            {this.examListItems()}
          </tbody>
        </table>
      </div>
    );
  }
}

export default connect(
  select,
  { canvasRequest, getTestingCentersAccount, ...ExamActions }
)(BaseExamList);
