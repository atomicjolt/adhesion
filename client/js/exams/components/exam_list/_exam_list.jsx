import React                         from 'react';
import { connect }                   from 'react-redux';
import _                             from 'lodash';
import appHistory                    from '../../../history';
import Defines                       from '../../defines';
import canvasRequest                 from '../../../libs/canvas/action';
import * as ExamActions              from '../../actions/exams';
import { listQuizzesInCourse }       from '../../../libs/canvas/constants/quizzes';
import ExamListItem                  from './exam_list_item';
import { getTestingCentersAccount }  from '../../actions/testing_centers_account';

const select = state => ({
  examList: state.exams.examList,
  lmsCourseId: state.settings.lmsCourseId,
  toolConsumerInstanceName: state.settings.toolConsumerInstanceName,
  lmsUserId: state.settings.lmsUserId,
  examRequests: state.exams.examRequests,
});

export class BaseExamList extends React.Component {

  static propTypes = {
    canvasRequest: React.PropTypes.func.isRequired,
    getTestingCentersAccount: React.PropTypes.func.isRequired,
    loadExamRequests: React.PropTypes.func.isRequired,
    lmsCourseId: React.PropTypes.string.isRequired,
    examList: React.PropTypes.shape({}),
    toolConsumerInstanceName: React.PropTypes.string.isRequired,
    lmsUserId: React.PropTypes.number.isRequired,
    examRequests: React.PropTypes.shape({}),
  }

  static goToExam(id) {
    appHistory.push(`exams/${id}`);
  }

  static tableHeader(styles) {
    return (
      <tr>
        <th style={{ ...styles.th, ...styles.larger }}>
          EXAM
        </th>
        <th style={{ ...styles.th, ...styles.smaller }}>
          STATUS
        </th>
      </tr>
    );
  }

  static getStyles() {
    return {
      header: {
        color: Defines.darkGrey,
        padding: '10px 20px',
      },
      hr: {
        borderTop: `2px solid ${Defines.lightGrey}`,
        margin: '0px',
      },
      table: {
        borderCollapse: 'collapse',
        width: '100%',
      },
      thead: {
        backgroundColor: Defines.lightBackground,
        color: Defines.lightText,
      },
      th: {
        fontWeight: 'normal',
        textAlign: 'left',
        padding: '15px 20px',
      },
      smaller: {
        width: '25%',
      },
      larger: {
        width: '75%',
      },
    };
  }

  componentWillMount() {
    const params = {
      course_id: this.props.lmsCourseId,
    };
    this.props.getTestingCentersAccount(this.props.toolConsumerInstanceName);
    this.props.canvasRequest(listQuizzesInCourse, params);
    this.props.loadExamRequests(this.props.lmsUserId);
  }

  examListItems() {
    return _.map(this.props.examList, exam => (
      <ExamListItem
        key={`exam_${exam.id}`}
        exam={exam}
        goToExam={id => BaseExamList.goToExam(id)}
        examRequest={this.props.examRequests[exam.id]}
      />
    ));
  }

  render() {
    const styles = BaseExamList.getStyles();
    return (
      <div>
        <h1 style={styles.header}>Request an Exam</h1>
        <hr style={styles.hr} />
        <table style={styles.table}>
          <thead style={styles.thead}>
            {BaseExamList.tableHeader(styles)}
          </thead>
          <tbody>
            {this.examListItems()}
          </tbody>
        </table>
      </div>
    );
  }
}

export default connect(
  select,
  { canvasRequest, getTestingCentersAccount, ...ExamActions }
)(BaseExamList);
