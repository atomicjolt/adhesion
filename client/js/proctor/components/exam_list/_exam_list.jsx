import React                         from 'react';
import { connect }                   from 'react-redux';
import _                             from 'lodash';
import appHistory                    from '../../../history';
import Defines                       from '../../defines';
import canvasRequest                 from '../../../libs/canvas/action';
import { listQuizzesInCourse }       from '../../../libs/canvas/constants/quizzes';
import ExamListItem                  from './exam_list_item';
import { getTestingCentersAccount }  from '../../actions/testing_centers_account';

const select = state => ({
  examList: state.exams.examList,
  lmsCourseId: state.settings.lmsCourseId,
  toolConsumerInstanceName: state.settings.toolConsumerInstanceName,
});

export class BaseExamList extends React.Component {

  static propTypes = {
    canvasRequest: React.PropTypes.func.isRequired,
    getTestingCentersAccount: React.PropTypes.func.isRequired,
    lmsCourseId: React.PropTypes.string.isRequired,
    examList: React.PropTypes.arrayOf(React.PropTypes.shape({})),
    toolConsumerInstanceName: React.PropTypes.string.isRequired,
  }

  static goToExam(id) {
    appHistory.push(`exams/${id}`);
  }

  static getStyles() {
    return {
      header: {
        color: Defines.darkGrey,
        padding: '10px 20px',
      },
      hr: {
        borderTop: `2px solid ${Defines.lightGrey}`,
        margin: '0px',
      },
      ul: {
        listStyle: 'none',
        padding: '0px',
        margin: '0px',
      }
    };
  }

  componentWillMount() {
    const params = {
      course_id: this.props.lmsCourseId,
    };
    this.props.getTestingCentersAccount(this.props.toolConsumerInstanceName);
    this.props.canvasRequest(listQuizzesInCourse, params);
  }


  examListItems() {
    return _.map(this.props.examList, exam => (
      <ExamListItem
        key={`exam_${exam.id}`}
        exam={exam}
        goToExam={id => BaseExamList.goToExam(id)}
      />
    ));
  }

  render() {
    const styles = BaseExamList.getStyles();
    return (
      <div>
        <h1 style={styles.header}>Choose an Exam to release</h1>
        <hr style={styles.hr} />
        <ul style={styles.ul}>
          {this.examListItems()}
        </ul>
      </div>
    );
  }
}

export default connect(select, { canvasRequest, getTestingCentersAccount })(BaseExamList);
