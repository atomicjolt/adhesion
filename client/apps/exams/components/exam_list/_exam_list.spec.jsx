import React from 'react';
import { shallow } from 'enzyme';
import { BaseExamList } from './_exam_list';

describe('Exam list', () => {
  let result;

  beforeEach(() => {
    const props = {
      canvasRequest: () => {},
      loadExamRequests: () => {},
      getTestingCentersAccount: () => {},
      lmsCourseId: '1',
      examList: [{ title: 'america', id: 1, access_code: 'proctored-exam-asdfasdf' }],
      examRequests: {
        1: {
          status: 'requested'
        }
      },
      lmsUserId: 'lms user id',
      toolConsumerInstanceName: 'instance name'
    };
    result = shallow(<BaseExamList {...props} />);
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });
});
