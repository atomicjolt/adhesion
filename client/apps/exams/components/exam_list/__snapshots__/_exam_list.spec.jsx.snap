// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exam list matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BaseExamList
    canvasRequest={[Function]}
    examList={
        Array [
            Object {
              "access_code": "proctored-exam-asdfasdf",
              "id": 1,
              "title": "america",
            },
          ]
    }
    examRequests={
        Object {
            "1": Object {
              "status": "requested",
            },
          }
    }
    getTestingCentersAccount={[Function]}
    lmsCourseId="1"
    lmsUserId="lms user id"
    loadExamRequests={[Function]}
    toolConsumerInstanceName="instance name"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          style={
                    Object {
                              "color": "#555555",
                              "padding": "10px 20px 0px 20px",
                            }
          }
>
          Request an Exam
</h1>,
        <div
          style={
                    Object {
                              "marginBottom": "20px",
                              "padding": "0px 20px",
                            }
          }
>
          If you need to change where you take an exam, create a new request.
</div>,
        <hr
          style={
                    Object {
                              "borderTop": "2px solid #ededed",
                              "margin": "0px",
                            }
          }
/>,
        <table
          className="qa-exam-list-table"
          style={
                    Object {
                              "borderCollapse": "collapse",
                              "width": "100%",
                            }
          }
>
          <thead
                    style={
                              Object {
                                        "backgroundColor": "#f5f5f5",
                                        "color": "#aaa",
                                      }
                    }
          >
                    <tr>
                              <th
                                        style={
                                                  Object {
                                                            "fontWeight": "normal",
                                                            "padding": "15px 20px",
                                                            "textAlign": "left",
                                                            "width": "50%",
                                                          }
                                        }
                              >
                                        EXAM
                              </th>
                              <th
                                        style={
                                                  Object {
                                                            "fontWeight": "normal",
                                                            "padding": "15px 20px",
                                                            "textAlign": "left",
                                                            "width": "25%",
                                                          }
                                        }
                              >
                                        TESTING CENTER
                              </th>
                              <th
                                        style={
                                                  Object {
                                                            "fontWeight": "normal",
                                                            "padding": "15px 20px",
                                                            "textAlign": "left",
                                                            "width": "25%",
                                                          }
                                        }
                              >
                                        STATUS
                              </th>
                    </tr>
          </thead>
          <tbody>
                    <examListItem
                              exam={
                                        Object {
                                                  "access_code": "proctored-exam-asdfasdf",
                                                  "id": 1,
                                                  "title": "america",
                                                }
                              }
                              examRequest={
                                        Object {
                                                  "status": "requested",
                                                }
                              }
                              goToExam={[Function]}
                    />
          </tbody>
</table>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Request an Exam",
          "style": Object {
            "color": "#555555",
            "padding": "10px 20px 0px 20px",
          },
        },
        "ref": null,
        "rendered": "Request an Exam",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "If you need to change where you take an exam, create a new request.",
          "style": Object {
            "marginBottom": "20px",
            "padding": "0px 20px",
          },
        },
        "ref": null,
        "rendered": "If you need to change where you take an exam, create a new request.",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "style": Object {
            "borderTop": "2px solid #ededed",
            "margin": "0px",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <thead
              style={
                            Object {
                                          "backgroundColor": "#f5f5f5",
                                          "color": "#aaa",
                                        }
              }
>
              <tr>
                            <th
                                          style={
                                                        Object {
                                                                      "fontWeight": "normal",
                                                                      "padding": "15px 20px",
                                                                      "textAlign": "left",
                                                                      "width": "50%",
                                                                    }
                                          }
                            >
                                          EXAM
                            </th>
                            <th
                                          style={
                                                        Object {
                                                                      "fontWeight": "normal",
                                                                      "padding": "15px 20px",
                                                                      "textAlign": "left",
                                                                      "width": "25%",
                                                                    }
                                          }
                            >
                                          TESTING CENTER
                            </th>
                            <th
                                          style={
                                                        Object {
                                                                      "fontWeight": "normal",
                                                                      "padding": "15px 20px",
                                                                      "textAlign": "left",
                                                                      "width": "25%",
                                                                    }
                                          }
                            >
                                          STATUS
                            </th>
              </tr>
</thead>,
            <tbody>
              <examListItem
                            exam={
                                          Object {
                                                        "access_code": "proctored-exam-asdfasdf",
                                                        "id": 1,
                                                        "title": "america",
                                                      }
                            }
                            examRequest={
                                          Object {
                                                        "status": "requested",
                                                      }
                            }
                            goToExam={[Function]}
              />
</tbody>,
          ],
          "className": "qa-exam-list-table",
          "style": Object {
            "borderCollapse": "collapse",
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <th
                                style={
                                                Object {
                                                                "fontWeight": "normal",
                                                                "padding": "15px 20px",
                                                                "textAlign": "left",
                                                                "width": "50%",
                                                              }
                                }
                >
                                EXAM
                </th>
                <th
                                style={
                                                Object {
                                                                "fontWeight": "normal",
                                                                "padding": "15px 20px",
                                                                "textAlign": "left",
                                                                "width": "25%",
                                                              }
                                }
                >
                                TESTING CENTER
                </th>
                <th
                                style={
                                                Object {
                                                                "fontWeight": "normal",
                                                                "padding": "15px 20px",
                                                                "textAlign": "left",
                                                                "width": "25%",
                                                              }
                                }
                >
                                STATUS
                </th>
</tr>,
              "style": Object {
                "backgroundColor": "#f5f5f5",
                "color": "#aaa",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <th
                    style={
                                        Object {
                                                            "fontWeight": "normal",
                                                            "padding": "15px 20px",
                                                            "textAlign": "left",
                                                            "width": "50%",
                                                          }
                    }
>
                    EXAM
</th>,
                  <th
                    style={
                                        Object {
                                                            "fontWeight": "normal",
                                                            "padding": "15px 20px",
                                                            "textAlign": "left",
                                                            "width": "25%",
                                                          }
                    }
>
                    TESTING CENTER
</th>,
                  <th
                    style={
                                        Object {
                                                            "fontWeight": "normal",
                                                            "padding": "15px 20px",
                                                            "textAlign": "left",
                                                            "width": "25%",
                                                          }
                    }
>
                    STATUS
</th>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "EXAM",
                    "style": Object {
                      "fontWeight": "normal",
                      "padding": "15px 20px",
                      "textAlign": "left",
                      "width": "50%",
                    },
                  },
                  "ref": null,
                  "rendered": "EXAM",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "TESTING CENTER",
                    "style": Object {
                      "fontWeight": "normal",
                      "padding": "15px 20px",
                      "textAlign": "left",
                      "width": "25%",
                    },
                  },
                  "ref": null,
                  "rendered": "TESTING CENTER",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "STATUS",
                    "style": Object {
                      "fontWeight": "normal",
                      "padding": "15px 20px",
                      "textAlign": "left",
                      "width": "25%",
                    },
                  },
                  "ref": null,
                  "rendered": "STATUS",
                  "type": "th",
                },
              ],
              "type": "tr",
            },
            "type": "thead",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <examListItem
                  exam={
                                    Object {
                                                      "access_code": "proctored-exam-asdfasdf",
                                                      "id": 1,
                                                      "title": "america",
                                                    }
                  }
                  examRequest={
                                    Object {
                                                      "status": "requested",
                                                    }
                  }
                  goToExam={[Function]}
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "exam_1",
                "nodeType": "function",
                "props": Object {
                  "exam": Object {
                    "access_code": "proctored-exam-asdfasdf",
                    "id": 1,
                    "title": "america",
                  },
                  "examRequest": Object {
                    "status": "requested",
                  },
                  "goToExam": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "tbody",
          },
        ],
        "type": "table",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            style={
                        Object {
                                    "color": "#555555",
                                    "padding": "10px 20px 0px 20px",
                                  }
            }
>
            Request an Exam
</h1>,
          <div
            style={
                        Object {
                                    "marginBottom": "20px",
                                    "padding": "0px 20px",
                                  }
            }
>
            If you need to change where you take an exam, create a new request.
</div>,
          <hr
            style={
                        Object {
                                    "borderTop": "2px solid #ededed",
                                    "margin": "0px",
                                  }
            }
/>,
          <table
            className="qa-exam-list-table"
            style={
                        Object {
                                    "borderCollapse": "collapse",
                                    "width": "100%",
                                  }
            }
>
            <thead
                        style={
                                    Object {
                                                "backgroundColor": "#f5f5f5",
                                                "color": "#aaa",
                                              }
                        }
            >
                        <tr>
                                    <th
                                                style={
                                                            Object {
                                                                        "fontWeight": "normal",
                                                                        "padding": "15px 20px",
                                                                        "textAlign": "left",
                                                                        "width": "50%",
                                                                      }
                                                }
                                    >
                                                EXAM
                                    </th>
                                    <th
                                                style={
                                                            Object {
                                                                        "fontWeight": "normal",
                                                                        "padding": "15px 20px",
                                                                        "textAlign": "left",
                                                                        "width": "25%",
                                                                      }
                                                }
                                    >
                                                TESTING CENTER
                                    </th>
                                    <th
                                                style={
                                                            Object {
                                                                        "fontWeight": "normal",
                                                                        "padding": "15px 20px",
                                                                        "textAlign": "left",
                                                                        "width": "25%",
                                                                      }
                                                }
                                    >
                                                STATUS
                                    </th>
                        </tr>
            </thead>
            <tbody>
                        <examListItem
                                    exam={
                                                Object {
                                                            "access_code": "proctored-exam-asdfasdf",
                                                            "id": 1,
                                                            "title": "america",
                                                          }
                                    }
                                    examRequest={
                                                Object {
                                                            "status": "requested",
                                                          }
                                    }
                                    goToExam={[Function]}
                        />
            </tbody>
</table>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Request an Exam",
            "style": Object {
              "color": "#555555",
              "padding": "10px 20px 0px 20px",
            },
          },
          "ref": null,
          "rendered": "Request an Exam",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "If you need to change where you take an exam, create a new request.",
            "style": Object {
              "marginBottom": "20px",
              "padding": "0px 20px",
            },
          },
          "ref": null,
          "rendered": "If you need to change where you take an exam, create a new request.",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "style": Object {
              "borderTop": "2px solid #ededed",
              "margin": "0px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <thead
                style={
                                Object {
                                                "backgroundColor": "#f5f5f5",
                                                "color": "#aaa",
                                              }
                }
>
                <tr>
                                <th
                                                style={
                                                                Object {
                                                                                "fontWeight": "normal",
                                                                                "padding": "15px 20px",
                                                                                "textAlign": "left",
                                                                                "width": "50%",
                                                                              }
                                                }
                                >
                                                EXAM
                                </th>
                                <th
                                                style={
                                                                Object {
                                                                                "fontWeight": "normal",
                                                                                "padding": "15px 20px",
                                                                                "textAlign": "left",
                                                                                "width": "25%",
                                                                              }
                                                }
                                >
                                                TESTING CENTER
                                </th>
                                <th
                                                style={
                                                                Object {
                                                                                "fontWeight": "normal",
                                                                                "padding": "15px 20px",
                                                                                "textAlign": "left",
                                                                                "width": "25%",
                                                                              }
                                                }
                                >
                                                STATUS
                                </th>
                </tr>
</thead>,
              <tbody>
                <examListItem
                                exam={
                                                Object {
                                                                "access_code": "proctored-exam-asdfasdf",
                                                                "id": 1,
                                                                "title": "america",
                                                              }
                                }
                                examRequest={
                                                Object {
                                                                "status": "requested",
                                                              }
                                }
                                goToExam={[Function]}
                />
</tbody>,
            ],
            "className": "qa-exam-list-table",
            "style": Object {
              "borderCollapse": "collapse",
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <tr>
                  <th
                                    style={
                                                      Object {
                                                                        "fontWeight": "normal",
                                                                        "padding": "15px 20px",
                                                                        "textAlign": "left",
                                                                        "width": "50%",
                                                                      }
                                    }
                  >
                                    EXAM
                  </th>
                  <th
                                    style={
                                                      Object {
                                                                        "fontWeight": "normal",
                                                                        "padding": "15px 20px",
                                                                        "textAlign": "left",
                                                                        "width": "25%",
                                                                      }
                                    }
                  >
                                    TESTING CENTER
                  </th>
                  <th
                                    style={
                                                      Object {
                                                                        "fontWeight": "normal",
                                                                        "padding": "15px 20px",
                                                                        "textAlign": "left",
                                                                        "width": "25%",
                                                                      }
                                    }
                  >
                                    STATUS
                  </th>
</tr>,
                "style": Object {
                  "backgroundColor": "#f5f5f5",
                  "color": "#aaa",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <th
                      style={
                                            Object {
                                                                  "fontWeight": "normal",
                                                                  "padding": "15px 20px",
                                                                  "textAlign": "left",
                                                                  "width": "50%",
                                                                }
                      }
>
                      EXAM
</th>,
                    <th
                      style={
                                            Object {
                                                                  "fontWeight": "normal",
                                                                  "padding": "15px 20px",
                                                                  "textAlign": "left",
                                                                  "width": "25%",
                                                                }
                      }
>
                      TESTING CENTER
</th>,
                    <th
                      style={
                                            Object {
                                                                  "fontWeight": "normal",
                                                                  "padding": "15px 20px",
                                                                  "textAlign": "left",
                                                                  "width": "25%",
                                                                }
                      }
>
                      STATUS
</th>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "EXAM",
                      "style": Object {
                        "fontWeight": "normal",
                        "padding": "15px 20px",
                        "textAlign": "left",
                        "width": "50%",
                      },
                    },
                    "ref": null,
                    "rendered": "EXAM",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "TESTING CENTER",
                      "style": Object {
                        "fontWeight": "normal",
                        "padding": "15px 20px",
                        "textAlign": "left",
                        "width": "25%",
                      },
                    },
                    "ref": null,
                    "rendered": "TESTING CENTER",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "STATUS",
                      "style": Object {
                        "fontWeight": "normal",
                        "padding": "15px 20px",
                        "textAlign": "left",
                        "width": "25%",
                      },
                    },
                    "ref": null,
                    "rendered": "STATUS",
                    "type": "th",
                  },
                ],
                "type": "tr",
              },
              "type": "thead",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <examListItem
                    exam={
                                        Object {
                                                            "access_code": "proctored-exam-asdfasdf",
                                                            "id": 1,
                                                            "title": "america",
                                                          }
                    }
                    examRequest={
                                        Object {
                                                            "status": "requested",
                                                          }
                    }
                    goToExam={[Function]}
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "exam_1",
                  "nodeType": "function",
                  "props": Object {
                    "exam": Object {
                      "access_code": "proctored-exam-asdfasdf",
                      "id": 1,
                      "title": "america",
                    },
                    "examRequest": Object {
                      "status": "requested",
                    },
                    "goToExam": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "tbody",
            },
          ],
          "type": "table",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
